ci/conan/all/conanfile.py

```diff
@@ -544,16 +512,18 @@ def generate(self):
         tc.variables["ARROW_ENABLE_TIMING_TESTS"] = False
         tc.variables["ARROW_BUILD_BENCHMARKS"] = False
         tc.variables["LLVM_SOURCE"] = "SYSTEM"
-        tc.variables["ARROW_WITH_UTF8PROC"] = self._with_utf8proc()
-        tc.variables["ARROW_BOOST_REQUIRED"] = self._with_boost()
+        tc.variables["ARROW_WITH_UTF8PROC"] = self.options.with_utf8proc
+        tc.variables["ARROW_BOOST_REQUIRED"] = self.options.with_boost
         tc.variables["utf8proc_SOURCE"] = "SYSTEM"
-        if self._with_utf8proc():
-            tc.variables["ARROW_UTF8PROC_USE_SHARED"] = bool(self.options["utf8proc"].shared)
+        if self.options.with_utf8proc:
+            tc.variables["ARROW_UTF8PROC_USE_SHARED"] = bool(self.dependencies["utf8proc"].options.shared)
         tc.variables["BUILD_WARNING_LEVEL"] = "PRODUCTION"
         if is_msvc(self):
-            tc.variables["ARROW_USE_STATIC_CRT"] = "MT" in str(self.settings.compiler.runtime)
-        if self._with_llvm():
-            tc.variables["LLVM_DIR"] = self.deps_cpp_info["llvm-core"].rootpath.replace("\\", "/")
+            tc.variables["ARROW_USE_STATIC_CRT"] = is_msvc_static_runtime(self)
+        if self.options.with_llvm:
+            tc.variables["LLVM_DIR"] = self.dependencies["llvm-core"].package_folder.replace("\\", "/")
+
+        tc.cache_variables["CMAKE_PROJECT_arrow_INCLUDE"] = os.path.join(self.export_sources_folder, "conan_cmake_project_include.cmake")
```

Comment:
The only difference from conan-center-index is this line: https://github.com/conan-io/conan-center-index/blob/master/recipes/arrow/all/conanfile.py#L504